synapse/storage/prepare_database.py

```diff
@@ -619,9 +619,10 @@ def _get_or_create_schema_state(
 
     txn.execute("SELECT version, upgraded FROM schema_version")
     row = txn.fetchone()
-    current_version = int(row[0]) if row else None
+    current_version = int(row[0]) if row is not None and len(row) > 0 else None
```

Comment:
(better to be explicit)

Reply:
Please revert this change, it is clearer to treat all false-y values the same.

Reply:
```suggestion
    current_version = int(row[0]) if row else None
```